scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -700,6 +700,24 @@ class ScioContext private[scio] (
   ): SCollection[U] =
     applyTransform(Option(name), root)
 
+  def transform[U](f: ScioContext => SCollection[U]): SCollection[U] = transform(this.tfName)(f)
+
+  def transform[U](name: String)(f: ScioContext => SCollection[U]): SCollection[U] =
+    wrap(transform_(name)(f(_).internal))
+
+  private[scio] def transform_[U <: POutput](f: ScioContext => U): U =
+    transform_(tfName)(f)
+
+  private[scio] def transform_[U <: POutput](name: String)(f: ScioContext => U): U = {
+    val sc: ScioContext = this
+    applyInternal(
+      name,
+      new PTransform[PBegin, U]() {
+        override def expand(pBegin: PBegin): U = f(sc)
```

Comment:
This line is kind of sketchy in that it *technically* throws away the supplied `pBegin` -- ideally we'd have a stable way to go from a `PInput` to its singleton enclosing `ScioContext`, but that's non-trivial...

@RustedBones , any ideas?